libr/bin/p/bin_qnx.c

```diff
@@ -0,0 +1,282 @@
+			RBinSection *ptr = NULL;
+			r_buf_fread_at (bf->buf, offset, (ut8 *) ldata, "ii", 1);
+			if (!(ptr = R_NEW0 (RBinSection))) {
+				return false;
+			}
+			ptr->name = strdup ("LMF_LOAD");
+			ptr->paddr = offset;
+			ptr->vaddr = ldata->offset;
+			ptr->vsize = lrec->data_nbytes - sizeof (lmf_data);
+			ptr->size = ptr->vsize;
+			ptr->add = true;
+		 	r_list_append (ret, ptr);
+		}
+        else if (lrec->rec_type == LMF_RW_END_REC) {
+			r_buf_fread_at (bf->buf, offset, (ut8 *) &qo->rwend, "ii", 1);
+		}
+        offset += lrec->data_nbytes;
+    }
+	sdb_ns_set (sdb, "info", qo->kv);
+	qo->sections = ret;
+	*bin_obj = qo;
+	return true;
+}
+
+static bool load(RBinFile *bf) {
+	const ut8 *byte = bf? r_buf_buffer (bf->buf): NULL;
+	ut64 size = bf? r_buf_size (bf->buf): 0;
+	if (!bf || !bf->o) {
+		return false;
+	}
+	return load_bytes (bf, &bf->o->bin_obj, byte, size, bf->o->loadaddr, bf->sdb);
+}
+
+/*
+ * Provides the info about the binary file
+ * @param RBinFile to extract the data from
+ * @return RBinInfo file with the info
+ */
+static RBinInfo *info(RBinFile *bf) {
+	RBinInfo *ret;
+	if (!bf || !bf->o || !bf->o->bin_obj) {
+		return NULL;
+	}
+	ret = R_NEW0 (RBinInfo);
+	if (!ret) {
+		return NULL;
+	}
+	ret->file = bf->file? strdup (bf->file): NULL;
+	ret->type = strdup ("QNX Executable");
+	ret->bclass = strdup ("qnx");
+	ret->machine = strdup ("i386");
+	ret->rclass = strdup ("QNX");
+	ret->arch = strdup ("x86");
+	ret->os = strdup ("any");
+	ret->subsystem = strdup ("any");
+	ret->lang = strdup ("C/C++");
+	ret->signature = true;
+	return ret;
+}
+
+static void header(RBinFile *bf) {
+	QnxObj *bin = bf->o->bin_obj;
+	struct r_bin_t *rbin = bf->rbin;
+
+	rbin->cb_printf ("QNX file header:\n");
+	rbin->cb_printf ("version						: 0x%xH\n", bin->lmfh.version);
+	rbin->cb_printf ("cflags						: 0x%xH\n", bin->lmfh.cflags);
+	rbin->cb_printf ("cpu						: 0x%xH\n", bin->lmfh.cpu);
+	rbin->cb_printf ("fpu						: 0x%xH\n", bin->lmfh.fpu);
+	rbin->cb_printf ("code_index					: 0x%xH\n", bin->lmfh.code_index);
+	rbin->cb_printf ("stack_index					: 0x%xH\n", bin->lmfh.stack_index);
+	rbin->cb_printf ("heap_index					: 0x%xH\n", bin->lmfh.heap_index);
+	rbin->cb_printf ("argv_index					: 0x%xH\n", bin->lmfh.argv_index);
+	rbin->cb_printf ("spare2[4]					: 0x0H\n", bin->lmfh.spare2);
+	rbin->cb_printf ("code_offset					: 0x%xH\n", bin->lmfh.code_offset);
+	rbin->cb_printf ("stack_nbytes					: 0x%xH\n", bin->lmfh.stack_nbytes);
+	rbin->cb_printf ("heap_nbytes					: 0x%xH\n", bin->lmfh.heap_nbytes);
+	rbin->cb_printf ("image_base					: 0x%xH\n", bin->lmfh.image_base);
+	rbin->cb_printf ("spare3[2]					: 0x0H\n");
+}
+
+/*
+ * No mention of symbols in the doc
+ */ 
+static RList* symbols(RBinFile *bf) {
+	return NULL;
+}
+
+// Returns the sections 
+static RList* sections(RBinFile *bf) {
+	QnxObj *qo = bf->o->bin_obj;
+	RList *ret = NULL;
+	if (!(ret = r_list_new ())) {
+		return NULL;
+	}
+	ret = r_list_clone (qo->sections);
```

Comment:
just `return r_list_clone (qo->sections);`

... are you sure you want to clone the list?

Reply:
If I don't clone the list, the destroy routine will cause a segmentation error.